test/Filter/File/RenameUploadTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace BsbFlysystemTest\Filter\File;
+
+use BsbFlysystem\Filter\File\RenameUpload;
+use BsbFlysystemTest\Framework\TestCase;
+use Prophecy\Argument;
+
+class RenameUploadTest extends TestCase
+{
+    protected $filesystem;
+
+    public function setUp()
+    {
+        $this->filesystem = $this->prophesize('League\Flysystem\FilesystemInterface');
+    }
+
+    public function testCanUploadFile()
+    {
+        $path = 'path/to/file.txt';
+        $this->filesystem->putStream($path, Argument::any())
```

Comment:
not sure about Argument::any() maybe better would be to at least check that type is resource? Unfortunately don't know how.